From b455138a72b55e72d3a1f32f97439b5568f9736f Mon Sep 17 00:00:00 2001
From: Isaiah Norton <isaiah@tiledb.io>
Date: Tue, 12 Nov 2019 16:14:25 -0500
Subject: [PATCH] Conda-only: patch capnproto build to work with anaconda

---
 cmake/Modules/FindCapnp_EP.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/Modules/FindCapnp_EP.cmake b/cmake/Modules/FindCapnp_EP.cmake
index c870d4ce..61b19b38 100644
--- a/cmake/Modules/FindCapnp_EP.cmake
+++ b/cmake/Modules/FindCapnp_EP.cmake
@@ -99,6 +99,7 @@ if (NOT CAPNP_FOUND)
           -DBUILD_TESTING=OFF
           "-DCMAKE_C_FLAGS=${CFLAGS_DEF}"
           "-DCMAKE_CXX_FLAGS=${CXXFLAGS_DEF}"
+          -DCMAKE_CXX_COMPILE_FEATURES=cxx_constexpr
           ${TILEDB_EP_BASE}/src/ep_capnp/c++
         UPDATE_COMMAND ""
         LOG_DOWNLOAD TRUE
-- 
2.20.1

