diff --git a/CMakeLists.txt b/CMakeLists.txt
index a19d4f907..fd3c46659 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -301,8 +301,9 @@ if (TILEDB_TESTS)
   add_dependencies(tests unit_array_schema unit_filter_create unit_filter_pipeline unit_metadata)
   add_dependencies(tests unit_compressors)
   add_dependencies(tests unit_range_subset)
-  add_dependencies(tests unit_mgc_dict)
-
+  if (ENABLE_MAGIC_TEST)
+    add_dependencies(tests unit_mgc_dict)
+  endif()
   add_subdirectory(test/regression)
 endif()
 
diff --git a/cmake/Modules/FindMagic_EP.cmake b/cmake/Modules/FindMagic_EP.cmake
index 7238bae8a..e7e47a967 100644
--- a/cmake/Modules/FindMagic_EP.cmake
+++ b/cmake/Modules/FindMagic_EP.cmake
@@ -90,7 +90,7 @@ if(NOT TILEDB_LIBMAGIC_EP_BUILT)
     ExternalProject_Add(ep_magic
       PREFIX "externals"
       GIT_REPOSITORY "https://github.com/TileDB-Inc/file-windows.git"
-      GIT_TAG "494060c2ea02494aabfbfc75e76d0af95c36156c"
+      GIT_TAG "1494ee365df9600fcc294ecf0592ebfa622b8713"
       GIT_SUBMODULES_RECURSE TRUE
       UPDATE_COMMAND ""
       CMAKE_ARGS
@@ -98,6 +98,7 @@ if(NOT TILEDB_LIBMAGIC_EP_BUILT)
         -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
         "-DCMAKE_C_FLAGS=${CFLAGS_DEF}"
         -Dlibmagic_STATIC_LIB=ON
+        -DDISABLE_MGC_CREATE=ON
       LOG_DOWNLOAD TRUE
       LOG_CONFIGURE TRUE
       LOG_BUILD TRUE
diff --git a/tiledb/sm/misc/CMakeLists.txt b/tiledb/sm/misc/CMakeLists.txt
index 4d851e757..ec426a01a 100644
--- a/tiledb/sm/misc/CMakeLists.txt
+++ b/tiledb/sm/misc/CMakeLists.txt
@@ -120,25 +120,27 @@ add_executable(compile_uuid EXCLUDE_FROM_ALL)
 target_link_libraries(compile_uuid PRIVATE uuid)
 target_sources(compile_uuid PRIVATE test/compile_uuid_main.cc)
 
-# simple unit test of magic.mgc embedded data vs external data
-find_package(Magic_EP REQUIRED)
-if(NOT EXISTS ${libmagic_DICTIONARY})
-  message(FATAL_ERROR "Failed to find libmagic 'magic.mgc' file.")
-endif()
-add_executable(unit_mgc_dict EXCLUDE_FROM_ALL)
-target_sources(unit_mgc_dict PRIVATE
-        test/unit_mgc_dict.cc 
-        mgc_dict.cc
-)
-target_link_libraries(unit_mgc_dict PRIVATE 
-  compressors
-  filter
-  libmagic
+if (ENABLE_MAGIC_TEST)
+  # simple unit test of magic.mgc embedded data vs external data
+  find_package(Magic_EP REQUIRED)
+  if(NOT EXISTS ${libmagic_DICTIONARY})
+    message(FATAL_ERROR "Failed to find libmagic 'magic.mgc' file.")
+  endif()
+  add_executable(unit_mgc_dict EXCLUDE_FROM_ALL)
+  target_sources(unit_mgc_dict PRIVATE
+          test/unit_mgc_dict.cc
+          mgc_dict.cc
   )
+  target_link_libraries(unit_mgc_dict PRIVATE
+    compressors
+    filter
+    libmagic
+    )
 
-target_compile_options(unit_mgc_dict PRIVATE -DTILEDB_PATH_TO_MAGIC_MGC=\"${libmagic_DICTIONARY}\")
-target_include_directories(unit_mgc_dict
-  PRIVATE
-    ${CMAKE_CURRENT_BINARY_DIR}
-   "${libmagic_INCLUDE_DIR}"
-)
+  target_compile_options(unit_mgc_dict PRIVATE -DTILEDB_PATH_TO_MAGIC_MGC=\"${libmagic_DICTIONARY}\")
+  target_include_directories(unit_mgc_dict
+    PRIVATE
+      ${CMAKE_CURRENT_BINARY_DIR}
+     "${libmagic_INCLUDE_DIR}"
+  )
+endif()
