diff --git a/cmake/Modules/FindMagic_EP.cmake b/cmake/Modules/FindMagic_EP.cmake
index 6779d349f..c1ff4da58 100644
--- a/cmake/Modules/FindMagic_EP.cmake
+++ b/cmake/Modules/FindMagic_EP.cmake
@@ -73,6 +73,9 @@ if (TILEDB_LIBMAGIC_EP_BUILT)
   )
 endif()
 
+set(BUILD_HOST_TOOLS_MACOS_UNIVERAL $ENV{BUILD_HOST_TOOLS_MACOS_UNIVERSAL})
+message(STATUS "BUILD_HOST_TOOLS_MACOS_UNIVERAL: $ENV{BUILD_HOST_TOOLS_MACOS_UNIVERSAL}")
+
 # if not yet built add it as an external project
 if(NOT TILEDB_LIBMAGIC_EP_BUILT)
   if (TILEDB_SUPERBUILD)
@@ -90,7 +93,7 @@ if(NOT TILEDB_LIBMAGIC_EP_BUILT)
     ExternalProject_Add(ep_magic
       PREFIX "externals"
       GIT_REPOSITORY "https://github.com/TileDB-Inc/file-windows.git"
-      GIT_TAG "5.38.1.tiledb"
+      GIT_TAG "5.38.2.tiledb"
       GIT_SUBMODULES_RECURSE TRUE
       UPDATE_COMMAND ""
       CMAKE_ARGS
@@ -98,7 +101,7 @@ if(NOT TILEDB_LIBMAGIC_EP_BUILT)
         -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
         "-DCMAKE_C_FLAGS=${CFLAGS_DEF}"
         -Dlibmagic_STATIC_LIB=ON
-        -DBUILD_MAGIC_MACOS_UNIVERSAL=$ENV{BUILD_MAGIC_MACOS_UNIVERSAL}
+        -DBUILD_HOST_TOOLS_MACOS_UNIVERSAL=${BUILD_HOST_TOOLS_MACOS_UNIVERSAL}
       LOG_DOWNLOAD TRUE
       LOG_CONFIGURE TRUE
       LOG_BUILD TRUE
